src/react-components/ui-root.js

```diff
@@ -399,17 +433,20 @@ class UIRoot extends Component {
     return !!this.state.micDevices.find(d => HMD_MIC_REGEXES.find(r => d.label.match(r)));
   };
 
+  micLabelForMediaStream = mediaStream => {
+    return (mediaStream && mediaStream.getAudioTracks().length > 0 && mediaStream.getAudioTracks()[0].label) || "";
+  };
+
   selectedMicLabel = () => {
-    return (
-      (this.state.mediaStream &&
-        this.state.mediaStream.getAudioTracks().length > 0 &&
-        this.state.mediaStream.getAudioTracks()[0].label) ||
-      ""
-    );
+    return this.micLabelForMediaStream(this.state.mediaStream);
+  };
+
+  micDeviceIdForMicLabel = label => {
+    return this.state.micDevices.filter(d => d.label === label).map(d => d.deviceId)[0];
   };
 
   selectedMicDeviceId = () => {
-    return this.state.micDevices.filter(d => d.label === this.selectedMicLabel).map(d => d.deviceId)[0];
```

Comment:
yes this was an ugly embarassing bug (no `()` on `selectedMicLabel`)

Reply:
I spent a bit of time trying Flow on this repo, but it was a hassle to make it understand AFrame and Three and it just spat out hundreds of errors, so it would take some effort to iron all of that out.